routing/datasource.go

```diff
@@ -96,6 +97,7 @@ func receiveFromClient(c DataClient, o Options, out chan<- *incomingData, quit <
 			continue
 		case initial || len(routes) > 0 || len(deletedIDs) > 0:
 			var incoming *incomingData
+			once.Do(firstLoad.Done)
```

Comment:
I think there is a race condition - this will signal first load before actual table is updated (which happens around `r.routeTable.Store(rt)` )

You can play/test this by adding time.Sleep here.

Reply:
Yup, tests are failing exactly because of this race.